docs/DeveloperGuide.md

```diff
@@ -1,38 +1,442 @@
-### Target user profile
+## [AddressBook Template](https://se-education.org/addressbook-level3/DeveloperGuide.html#proposed-undoredo-feature)
 
-{Describe the target user profile}
+The format of this Developer’s guide (DG) as well as some basic support is inspired/ taken from the AddressBook 
+project’s DG.
 
-### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+## [Bitwarden](https://go.bitwarden.com/password-management-for-business-teams-organizations/?utm_source=google&utm_medium=cpc&utm_campaign=AW_APAC_NU_CL_Bitwarden_en_GSN_DTMB_Brand_KW:Brand_Exact&utm_content=646898599878&utm_term=bitwarden&hsa_acc=2567950947&hsa_cam=11319090405&hsa_grp=114652450007&hsa_ad=646898599878&hsa_src=g&hsa_tgt=kwd-442029814536&hsa_kw=bitwarden&hsa_mt=b&hsa_net=adwords&hsa_ver=3&gclid=CjwKCAjwzuqgBhAcEiwAdj5dRn0X9IDtuPlHiXveE0H8FMOOYNtwMBxX7lUL4yzK8zUTKH_58EV-4xoCrVkQAvD_BwE)
 
-## User Stories
+The idea for this product stems out from Applications like Bitwarden and the like that serve to store and protect users passwords. Many of our features were inspired by these applications.
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
 
-## Non-Functional Requirements
+## CLI Applications
 
-{Give non-functional requirements}
+The idea for this product is for users who are familiar with CLI environments. Users are able to see only the necessary information given in text format. Things can be done faster with just a single line of input command compared to navigating through a GUI environment.
 
-## Glossary
 
-* *glossary item* - Definition
+---
 
-## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+# Setting up, getting started
+
+
+## Setting up the project in your computer
+
+First, fork this repo, and clone the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+
+
+
+1. **Configure the JDK:** Follow the guide [[se-edu/guides] IDEA: Configuring the JDK](https://se-education.org/guides/tutorials/intellijJdk.html) to ensure Intellij is configured to use JDK 11.
+2. **Import the project as a Gradle project:** Follow the guide [[se-edu/guides] IDEA: Importing a Gradle project](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA. \
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ExclamationMark.png" width="4%" />
+Note: Importing a Gradle project is slightly different from importing a normal Java project.
+3. **Verify the setup:**
+    1. Run the _seedu.address.Main_ and try a few commands.
+    2. [Run the tests](https://se-education.org/addressbook-level3/Testing.html) to ensure they all pass.
```

Comment:
Perhaps the indentation can be done better? So that it will be more reader friendly.